src/main/java/net/imagej/convert/ROITreeToRealMaskRealPredicates.java

```diff
@@ -0,0 +1,62 @@
+package net.imagej.convert;
+
+import net.imagej.roi.ROITree;
+import net.imglib2.roi.geom.real.SuperEllipsoid;
+import org.scijava.convert.AbstractConverter;
+import org.scijava.convert.Converter;
+import org.scijava.plugin.Plugin;
+import org.scijava.util.TreeNode;
+import org.scijava.util.Types;
+
+import java.lang.reflect.Type;
+
+public class ROITreeToRealMaskRealPredicates
+{
+
+
+	/**
+	 * Converter from ROITree to SuperEllipsoid
+	 * @author Gabriel Selzer
+	 */
+	@Plugin(type = Converter.class)
+	public static class ConvertIntArrayToFinalInterval //
+			extends AbstractConverter<ROITree, SuperEllipsoid> //
+	{
+		@Override
+		public boolean canConvert(final Object src, final Type dest) {
+			return canConvert( src, Types.raw(dest));
+		}
+
+		@Override
+		public boolean canConvert(final Object src, final Class<?> dest) {
+			// Assert that src is a ROITree and dest SuperEllipsoid
+			if (!super.canConvert( src, dest)) return false;
+			ROITree srcTree = (ROITree) src;
+			// Assert exactly one ROI in the tree
+			if (srcTree.children().size() != 1) return false;
```

Comment:
With Ops v1, we allowed ops to match contingently on the *content* of an object, rather than only its *type*. With Ops v2, we are no longer allowing that, in favor of matches happening on *types only*.

Analogously, with converters, while you *can* restrict conversion based on the content of an object rather than only its source type... I am wondering if we should really be leaning on this so heavily. There are places in the SciJava codebase that ask if a certain conversion *from type to type* can be done, and for this `ROITree` &rarr; `SuperEllipsoid` (or whatever `RealMask` subtype) conversion, we will be forced to respond YES. But then when you actually try to convert your `ROITree` object which does not in fact contain a single `SuperEllipsoid` instance, we will throw `ClassCastException`. This is bad.

I was hoping we could move away from contingently converting based on object contents, in favor of only types. Do you see a way forward here? Why do we need converters for `ROITree` to the various `RealMask` kinds, anyway?

Reply:
> Do you see a way forward here?

Sure. What do you think about:

```java
@Functional
public interface Converter<T, U> extends Function<T, Optional<U>> {
  public Optional<U> convert(final T t);

  @Override
  default Optional<U> apply(final T t) { return convert(t);}

  default boolean canConvert(final T t) {return false;} // default implementation
}
```

I wonder if we'd see any performance impact from that `Optional` if conversion always worked (i.e. was not conditional)